core/src/replay_stage.rs

```diff
@@ -212,51 +228,57 @@ impl ReplayStage {
                     );
 
                     let ancestors = Arc::new(bank_forks.read().unwrap().ancestors());
-                    let votable = Self::generate_votable_banks(
-                        &ancestors,
-                        &bank_forks,
-                        &tower,
-                        &mut progress,
-                    );
-
-                    if let Some((_, bank, _, total_staked)) = votable.into_iter().last() {
-                        subscriptions.notify_subscribers(bank.slot(), &bank_forks);
-
-                        if let Some(votable_leader) =
-                            leader_schedule_cache.slot_leader_at(bank.slot(), Some(&bank))
-                        {
-                            Self::log_leader_change(
-                                &my_pubkey,
-                                bank.slot(),
-                                &mut current_leader,
-                                &votable_leader,
-                            );
+                    loop {
```

Comment:
Is this a loop so that you will vote on all the ancestors, the heaviest bank, and then the children?

If there are children, `vote_bank` will skip directly to the child and skip voting for the ancestors +the heaviest bank

Reply:
@carllin If we switch forks post lockup this will let the validator catch up.